src/test/java/com/haxwell/apps/quizki/services/QuestionServiceImplTest.java

```diff
@@ -0,0 +1,278 @@
+		topic1 = new Topic(topicStr1);
+		topic1.setId(1l);
+
+		
+		reference1 = new Reference(refStr1);
+		reference1.setId(1l);
+		
+		Optional<User> userOpt = Optional.of(user);
+	
+		when(userRepo.findById(userId)).thenReturn(userOpt);
+		
+		when(topicRepo.findByText(topicStr1)).thenReturn(topic1);
+		when(topicRepo.findByText(topicStr2)).thenReturn(null);
+		
+		when(refRepo.findByText(refStr1)).thenReturn(reference1);
+		when(refRepo.findByText(refStr2)).thenReturn(null);
+		
+		//Stubbing the save method inserts ID value 2 in the (null) Topics & Refs and sets ids in Question parameter & choices
+		//to emulate JPA cascading in the returned Question
+		//Note: Both the topics and references returned here will not have Questions which are not relevant to outputDTO
+		
+		when(questionRepo.save(any(Question.class))).thenAnswer(new Answer<Question>() {
+ 
+            public Question answer(InvocationOnMock invocation) throws Throwable {
+                Question question = invocation.getArgument(0);
+                question.setId(1l);
+                
+                for(Topic t: question.getTopics()) {
+                	if(t.getId() == null)
+                		t.setId(2l);
+                }
+                
+                for(Reference r: question.getReferences()) {
+                	if(r.getId() == null)
+                		r.setId(2l);
+                }
+                
+                int i = 1;
+                for(Choice c: question.getChoices()) {
+                	c.setId(i);
+                	++i;
+                }
+
+                return question;
+            }
+        });
+		
+		assertNotNull(qsImpl);
+		
+		try {
+			outputDTO = qsImpl.createQuestion(inputDTO);
+		} catch (CreateQuestionDTOException e) {
+			System.out.println("!!!!!!!!!!!!!!!!------>>>>>" + e.getMessage());
+		}
+		
+		
+		verify(userRepo).findById(userId);
+		verify(topicRepo).findByText(topicStr1);
+		verify(topicRepo).findByText(topicStr2);
+		verify(refRepo).findByText(refStr1);
+		verify(refRepo).findByText(refStr2);
+		verify(questionRepo).save(any(Question.class));
+		
+
+		
+		assertThat(outputDTO.getId(), equalTo(1l));
+		assertThat(outputDTO.getUserId(), equalTo(userId));
+		assertThat(outputDTO.getText(), equalTo(text));
+		assertThat(outputDTO.getDescription(), equalTo(description));
+		assertThat(outputDTO.getQuestionType(), equalTo(type));
+		assertThat(outputDTO.getDifficulty(), equalTo(difficulty));
+		
+		assertThat(outputDTO.getChoices(), hasSize(2));
+		assertThat(outputDTO.getTopics(), hasSize(2));
+		assertThat(outputDTO.getReferences(), hasSize(2));
+		
+	}
+	
+	@Test
+	public void getQuestionByIdTest() {
+		
+		question = getQuestion(1l);
+		
+		Optional<Question> inQuestionOpt = Optional.of(question);
+		
+		when(questionRepo.findById(1l)).thenReturn(inQuestionOpt);
+		
+		try {
+			outputDTO = qsImpl.getQuestionById(Long.toString(1l));
+		} catch (GetQuestionException e) {
+			System.out.println(e.getLocalizedMessage());
+		}
```

Comment:
It's better to `fail()` here. If we get an unexpected Exception, we want to be loud about it.